File: 02-week-2--recursion/02-day-2--recursive-search/javascript/recursive_search.js
function recursiveSearch(arr, target) {
  // type your code here
  // iterative
  // if (arr.length > 0) { 
  //   if (arr.find(element => element == target)){
  //     return true
  //   } else {
  //     return false
  //   }
  // } else {
  //   return false
  // }

  while (arr.length > 0) {
    if (arr.find(element => element == target)){
      return true
    } else {
      recursiveSearch(arr.shift(), target)
    }
  }
    if (arr.length == 0){
      return false
  }
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: true");
  console.log("=>", recursiveSearch([1, 2, 3], 2));

  console.log("");

  console.log("Expecting: false");
  console.log("=>", recursiveSearch([3, 2, 1], 4));
}

module.exports = recursiveSearch;

// Please add your pseudocode to this file
// arr is an array , target is a number that either exists or doesnt in arr
// iterate over the array and see if that element equals target
// no duplicates
// And a written explanation of your solution
// if an element matches the target, break the loop and return true
// otherwise, remove that element from the array and work on a smaller array
// if the array is empty, there wasnt a match
// therefore, return true

File: 02-week-2--recursion/02-day-2--recursive-search/ruby/recursive_search.rb
def recursive_search(arr, target)
  # type your code in here
  # if arr.length > 0
  if arr.empty? 
    return false
  else
    if arr.first == target
      return true
    else 
      arr.shift
      recursive_search(arr, target)
    end
  end
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: true"
  puts "=>", recursive_search([1, 2, 3], 2)

  puts

  puts "Expecting: false"
  puts "=>", recursive_search([3, 2, 1], 4)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
# if the array is empty, immediately break the loop and return false
# otherwise, see if the first element is the same as the target.
# if it's not, remove that element and try again with the new first element
